### PR TITLE
[jquery] Update return type of children method

### PR DESCRIPTION
**Pull request by martin-badin**
Discussion: #57409

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://api.jquery.com/children/#children-selector>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by Steve-Fenton** [comment]
Hi @typescript-bot 

Thanks for the heads-up but I'm not eligible to review this, even though you tagged me. I think some rules have changed and even though I'm an owner the package now needs a DT maintainer to approve it.

Thanks

:)

**Comment by sandersn** [comment]
@Steve-Fenton (and others that the bot pinged): I do need some owners to review and OK the change even if github doesn't technically allow you to approve it. The discussion leading to this change makes sense, so once one or two owners OK it, I'll merge it.

**Comment by danielrentz** [comment]
I would expect this to work similarly to the find method, for example, `$something.children("span")` should result in a `JQuery<HTMLSpanElement>`. The `find` method is defined as

```
    find<K extends keyof HTMLElementTagNameMap>(selector_element: K | JQuery<K>): JQuery<HTMLElementTagNameMap[K]>;
    find<K extends keyof SVGElementTagNameMap>(selector_element: K | JQuery<K>): JQuery<SVGElementTagNameMap[K]>;
    find<E extends HTMLElement>(selector_element: JQuery.Selector | Element | E | JQuery<E>): JQuery<E>;
```

**Comment by martin-badin** [comment]
You are right. I will update my PR.

**Comment by amcasey** [comment]
@danielrentz Do things work as you expect now?

**Comment by danielrentz** [comment]
@amcasey Yes looks good. Thanks!

**Comment by amcasey** [comment]
As @sandersn said, the change LGTM, but I'm reluctant to merge it without hearing from at least one owner, given how widely used this package is.

@diegovilar @thorn0 @calebstdenis @leonard-thieu @scipper @peterblazejewicz @borisyankov @choffmeister @Steve-Fenton @Diullei @tasoili @jasons-novaleaf @seanski @Guuz @ksummerlin @basarat @nwolverson @derekcicerone @AndrewGaspar @seikichi @benjaminjackman @s093294 @JoshStrobl @johnnyreilly @DickvdBrink @King2500 @terrymun @princefishthrower

File: types/jquery/JQuery.d.ts
Comment (peterblazejewicz) [inline types/jquery/JQuery.d.ts:1785]:
```suggestion
    children(selector?: JQuery.Selector): this;
```
to keep with convention in file?

Comment (peterblazejewicz) [reply types/jquery/JQuery.d.ts:1785]: oh, sorry, I misread the intent/fixed topic.

Comment (martin-badin) [reply types/jquery/JQuery.d.ts:1785]: no problem. Is it fine?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**
